pkg/dockerutil/dockerutils.go

```diff
@@ -487,3 +495,31 @@ func RunSimpleContainer(image string, name string, cmd []string, entrypoint []st
 
 	return stdout.String(), nil
 }
+
+// ImageExistsLocally determines if an image is available locally.
+func ImageExistsLocally(imageName string) (bool, error) {
+	client := GetDockerClient()
+
+	images, err := client.ListImages(docker.ListImagesOptions{
+		Filter: imageName,
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(images) == 0 {
+		return false, nil
+	}
+
+	for _, i := range images {
```

Comment:
I don't think we need this stanza below at all? It seems like the filter already got us the exact one and only image we want, at least that's what it looks like to me stepping through. If I'm not mistaken we can "if we found one, true, if we didn't, false"